projects/scion/microfrontend-platform/src/lib/host/manifest-registry/ɵmanifest-registry.ts

```diff
@@ -99,15 +99,23 @@ export class ɵManifestRegistry implements ManifestRegistry, PreDestroy {
    */
   private hasIntentionForCapability(appSymbolicName: string, capability: Capability): boolean {
     const filter: ManifestObjectFilter = {appSymbolicName, type: capability.type, qualifier: capability.qualifier};
-    return this._intentionStore.find(filter, intentionQualifier => new QualifierMatcher(intentionQualifier, {evalAsterisk: true, evalOptional: true}).matches(capability.qualifier)).length > 0;
+    return this._intentionStore.find(filter, intentionQualifier => new QualifierMatcher(intentionQualifier, {evalAsterisk: true}).matches(capability.qualifier)).length > 0;
   }
 
   public async registerCapability(capability: Capability, appSymbolicName: string): Promise<string> {
     if (!capability) {
       throw Error('[CapabilityRegisterError] Missing required capability.');
     }
-    if (capability.qualifier && capability.qualifier.hasOwnProperty('*')) {
-      throw Error('[CapabilityRegisterError] Asterisk wildcard (\'*\') not allowed in the qualifier key.');
+    if (capability.qualifier) {
+      if (capability.qualifier.hasOwnProperty('*')) {
+        throw Error('[CapabilityRegisterError] Asterisk wildcard (\'*\') not allowed in the qualifier key.');
+      }
+      if (Object.values(capability.qualifier).some(value => value === '*')) {
+        throw Error('[CapabilityRegisterError] Asterisk wildcard (\'*\') not allowed in the qualifier value. Use required params instead.');
+      }
+      if (Object.values(capability.qualifier).some(value => value === '?')) {
```

Comment:
Should we remove the check for the optional wildcard in a future release? Then, I suggest creating an issue and putting a TODO here.

Reply:
done